src/s1reader/s1_burst_slc.py

```diff
@@ -423,6 +423,55 @@ def as_dict(self):
             self_as_dict[key] = val
         return self_as_dict
 
+    def bistatic_delay(self, xstep=1, ystep=1):
+        '''Computes the bistatic delay correction as
+        described in Gisinger et al, 2021
+
+        Parameters
+        ----------
+        xstep : int
+           spacing along x direction (range direction) in units of pixels
+
+        ystep : int
+           spacing along y direction (azimuth direction) in units of pixels
+
+
+        Returns
+        --------
+           LUT2D object of bistatic delay correction in seconds as a function
+           of the range and zimuth indices. This correction needs to be added
+           to the SLC tagged azimuth time to get the corrected azimuth times.
+        '''
+
+        pri = 1.0 / self.prf_raw_data
+        tau0 = self.rank * pri
+
+        mid_range = self.starting_range + 0.5 * self.width * self.range_pixel_spacing
+        tau_mid = mid_range * 2.0 / isce3.core.speed_of_light
+
+        nx = int(self.width / xstep)
+        ny = int(self.length / ystep)
+
+        x = np.arange(0, (nx+1)*xstep, xstep, dtype = int)
+        y = np.arange(0, (ny+1)*ystep, ystep, dtype = int)
+        x_mesh = np.meshgrid(x, y)[0]
+        slant_range = self.starting_range + x_mesh * self.range_pixel_spacing
+        tau = slant_range * 2.0 / isce3.core.speed_of_light
+
+        # the first term (tau_mid/2) is the bulk bistatic delay which was
+        # removed from the orginial azimuth time by the ESA IPF. Based on
+        # Gisinger et al, 2021, ESA IPF has used the mid of the second subswath
+        # to compute the bulk bistatic delay. However currently we have not
+        # been able to verify this from ESA documents. In this implementation
+        # we have used the range to the middle of the burst of interest to
+        # compute the bulk bistatic delay. The correction can be potentially
+        # biased by the amount of discrepancy between the ESA IPF bulk bistatic
+        # correction and our assumed middle of the burst of interest.
+        bistatic_correction = tau_mid / 2 + tau / 2 - tau0
```

Comment:
```suggestion
        nx = np.ceil(self.width / xstep)
        ny = np.ceil(self.length / ystep)

        x = np.arange(0, nx*xstep, xstep, dtype=int)
        y = np.arange(0, ny*ystep, ystep, dtype=int)

        slant_range = self.starting_range + x * self.range_pixel_spacing
        tau = slant_range * 2.0 / isce3.core.speed_of_light

        # the first term (tau_mid/2) is the bulk bistatic delay which was
        # removed from the orginial azimuth time by the ESA IPF. Based on
        # Gisinger et al, 2021, ESA IPF has used the mid of the second subswath
        # to compute the bulk bistatic delay. However currently we have not
        # been able to verify this from ESA documents. In this implementation
        # we have used the range to the middle of the burst of interest to
        # compute the bulk bistatic delay. The correction can be potentially
        # biased by the amount of discrepancy between the ESA IPF bulk bistatic
        # correction and our assumed middle of the burst of interest.
        bistatic_correction_vec = tau_mid / 2 + tau / 2 - tau0
        bistatic_correction = np.stack([bistatic_correction_vec for _ in range(ny)],
                                       axis=0)
```
Since all the lines are the same, this reduces amount of computation and memory usage. Also nits for no spaces in named parameter assignment in `np.arange`.